---
title: 'GWAS for the Pea Single Plant Plus Collection (PSPPC), see traits below'
author: "Carolina BT"
date: "March 2025"
Script: 
output:
  html_document: default
  pdf_document: default
  chunk_output_type: console
chunk_output_type: console
editor_options: 
  chunk_output_type: console
---

# ##############################

## TRAITS

Moisture (%)                            = Moisture_Per
Protein (unadjusted)                    = Protein_unadjusted
SAA (unadjusted)                        = SAA_unadjusted
Dietary Fiber (unadjusted)              = Dietary_Fiber_unadjusted
Total Starch (unadjusted)               = Total_Starch_unadjusted
Resistance Starch (unadjusted)          = Resistance_Starch_unadjusted
Protein (15% DB Moisture)               = Protein_15per_DB_Moisture
SAA (15% DB Moisture)                   = SAA_15per_DB_Moisture
Dietary Fiber (15% DB Moisture)         = Dietary_Fiber_15per_DB_Moisture
Total Starch (15% DB Moisture)          = Total_Starch_15per_DB_Moisture
Resistance Starch (15% DB Moisture)     = Resistance_Starch_15per_DB_Moisture
Protein digestibility (%)               = Protein_digestibility_per

## FOLDER

/project/dthavar/dilthavar/acballe/PSPPC_GWAS/

# #################################################
# Set working directory and load libraries for GWAS

``` {r, echo=FALSE}

#clear environment
rm(list = ls())

print("#------------------------------------Set working directory----------------------------------------#")
setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/")
getwd()

print("#------------------------------------Install and call packages------------------------------------#")

library(GAPIT)
library(dplyr)
library(plyr) #ldply
library(tidyverse)
library(CMplot)
library(ggplot2)
library(data.table) #fread
library(ggpubr) #ggarrange
library(gt) #tab_style
library(qqman) #manhattan function
library(CMplot) #circular manhattan plot
library(sommer)
library(cowplot) #plot_grid
library(stringr)

```

```{r, echo=TRUE}
###############################################################################################################
```

# ############################################################################################
### Read files

```{r, include=FALSE}

print("#---------------------------------Phenotype BLUPS---------------------------------------#")
myY_blups <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_Phenotypes/PSP3_BLUPs.csv", head = TRUE)
myY_blups <- as.data.frame(myY_blups)
str(myY_blups)
myY_blups[1:2,]
dim(myY_blups)

print("#---------------------------------Phenotype BAYESREE---------------------------------------#")
#116 traits (12 traits x 4 years (all, 20+21, 20, 21) + 10 x 1 year (2202) x 2 models (with and without block))
myY_bayes <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_Phenotypes/PSP3_BAYESREE.csv", head = TRUE)
myY_bayes <- as.data.frame(myY_bayes)
str(myY_bayes)
myY_bayes[1:2,]
dim(myY_bayes)

print("#-----------------------------------Genotype---------------------------------------#")
myG <- read.table("/project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.hmp.txt", head = FALSE) #calculated in Tassel GUI, and removed "#" manually in column names
#myG <- read.table("/project/dthavar/dilthavar/njohns9/PSPProteinGWAS/VCFJunk/PSPPC.filt.hmp.txt", head = FALSE) 
myG[1:5,1:15]
dim(myG)

#table(myY$Taxa %in% colnames(myG_pea)) #check overlapping names between phneo and geno, run this if head = T, in read.table of hapmap
#table(colnames(myG_pea) %in% myY$Taxa) #check overlapping names between phneo and geno, run this if head = T, in read.table of hapmap

print("#---------------------------------READ kinship---------------------------------------#")
#generated by GAPIT with VanRaden METHOD
myKI_VR <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GAPIT.Genotype.Kin_VanRaden.csv", header = F)
myKI_VR[1:2,]
dim(myKI_VR)
#generated by GAPIT with VanRaden METHOD
myKI <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GAPIT.Genotype.Kin_Zhang.csv", header = F)
myKI[1:2,]
dim(myKI)

```

```{r, echo=TRUE}
###############################################################################################################
```

# ############################################################################################
### Run GAPID (below information is taken from the manual)

single locus model category, MLM is superior to GLM, CMLM is superior to MLM, ECMLM is superior to CMLM, SUPER and MLM(?) are superior to MLM

      1. Generalized Linear Model (GLM)
      2. Mixed linear model (MLM or Q+K)
      3. Compressed MLM (CMLM)
      4. Enriched CMLM (ECMLM)
      5. Settlement of MLM Under Progressively Exclusive Relationship (SUPER)

Multiple loci models (e.g. MLMM, FarmCPU, and BLINK) are superior to the rest. Within multiple loci model category, FarmCPU is superior to MLMM and BLINK is superior to FarmCPU

      6. Multiple Loci Mixed Model (MLMM)
      7. Fixed and random model Circulating Probability Unification (FarmCPU)
      8. Bayesian-information and Linkage-disequilibrium Iteratively Nested Keyway (BLINK).

# incorporate population structure as a covariate (fixed effect). Principal coordinates analysis (PCA) is a common method to do this and is easily incorporated in GAPIT.
# Since we can use results from PCA as a covariate, how many PCs should we decide to use? Each PC explains a proportion of the cumulative variation. Typically, you could select the amount of PCs that explain ~25% or 50% of the variation.

# GAPIT_all_MODELS <- GAPIT(Y=myY, G=myG, PCA.total = 5, model=c("GLM", "MLM", "CMLM", "ECMLM", "SUPER", "MLMM", "FarmCPU", "Blink"), Multiple_analysis = TRUE)
# GAPIT_all_MODELS <- GAPIT(Y=myY, G=myG, PCA.total = 5, model=c("GLM", "MLM", "CMLM", "ECMLM", "SUPER", "MLMM", "FarmCPU", "Blink"), Multiple_analysis = TRUE, kinship.algorithm = "VanRaden") #VanRaden is default
# GAPIT_all_MODELS <- GAPIT(Y=myY, G=myG, PCA.total = 5, model=c("GLM", "MLM", "CMLM", "ECMLM", "SUPER", "MLMM", "FarmCPU", "Blink"), SNP.MAF = 0.01, SNP.FDR = 0.05, Multiple_analysis = TRUE, kinship.algorithm = "VanRaden")
# 
# #If eigenvalues contain zeros or negatives, your kinship matrix might need fixing.
# eigenvalues <- read.csv("GAPIT.Genotype.PCA_eigenvalues.csv", head = TRUE, sep = ",")
# print(eigenvalues$x)
# hist(eigenvalues$x)

# ############################################################################################
#run GWAS wiuth blups (DID NOT DO!!!!)

```{r, echo=FALSE}

print("#---------------------------------View data BLUPS----------------------------------#")

#Check data
myY_blups[1:2,]
myG[1:5,1:12]

print("#-------------------------------Run GAPIT with BLUPS-------------------------------#")

#make variables
#traits <- c("Moisture", "Protein", "SAA", "T_starch", "R_starch", "PDg")
traits <- c("Protein")
traits

for (i in traits){
  print(paste0("Processing trait: ", i))
  #setwd
  setwd(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_BLUPs/",i, sep=""))
  print(getwd())
  #select trait column 
  data <- myY_blups %>% select(matches("Taxa") | matches(i))
  print(data[1:2,])
  #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("GLM", "MLM", "CMLM", "ECMLM", "SUPER", "MLMM", "FarmCPU", "Blink"), Multiple_analysis = TRUE, kinship.algorithm = "VanRaden") 
  #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("GLM", "MLMM", "Blink"), Multiple_analysis = TRUE, kinship.algorithm = "VanRaden") 
}

```

```{r, echo=TRUE}
###############################################################################################################
```

______________________ Run GWAS in palmetto 
sbatch GAPIT_modeSelection.sbatch
squeue --me
                      
                      ______________________ GAPIT_modeSelection.sbatch
                      #!/bin/bash
                                            
                      #SBATCH --job-name GAPIT_modeSelection
                      #SBATCH --nodes 1
                      #SBATCH --tasks-per-node 1
                      #SBATCH --cpus-per-task 16
                      #SBATCH --mem 100gb
                      #SBATCH --time 72:00:00
                      #SBATCH --mail-type BEGIN
                      #SBATCH --mail-type END
                      #SBATCH --mail-type FAIL
                      #SBATCH --output=GAPIT_modeSelection.out
                      #SBATCH --error=GAPIT_modeSelection.err
                                            
                      cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS
                                            
                      module load r/4.4.1
                      Rscript GAPIT_modeSelection.R
                      ______________________
______________________

# model selection

all trait in /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/model_selection/

look for file *Model.Selection.Results.csv

check Bayesian Information Criterion (BIC) and Log-Likelihood (LL) values for different numbers of Principal Components (PCs) or covariates

The model with the PC that has the highest BIC, suggest it provides the best fit with the least complexity.


I got this:

The upper bound of groups (group.to) is not sufficient. both boundries were set to a and GLM is performed!
The upper bound of groups (group.to) is not sufficient. both boundries were set to their minimum and GLM is performed!

which means 
This message means that GAPIT could not form the intended kinship groups, so it defaulted to running GLM (General Linear Model) instead of MLM (Mixed Linear Model) or CMLM (Compressed MLM).
Without a kinship matrix, GAPIT defaults to GLM, which ignores population structure.

Add kinship created by GAPIT in previous runs (deleted)

    I tried adding "group.to = 267, and 10 (subpopulation, I added kinship and is not working)

```{r, echo=FALSE}
print("#---------------------------------View data BAYES----------------------------------#")

#Check data
myY_bayes[1:2,]
myG[1:5,1:12]
myKI_VR[1:5,1:12]

#check missing values
sum(is.na(myY_bayes))   # Count missing phenotype values
sum(is.na(myG))  # Count missing genotype values

print("#-------------------------------TEST: Run GAPIT with BAYES, for model selection-------------------------------#")

                  #make variables
                  traits <- c("Protein")
                  traits
                  # this is for testing
                  for (i in traits){
                    print("#############################################")
                    print(paste0("Processing trait: ", i))
                    #setwd
                    setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/model_selection/")
                    print(getwd())
                    #select trait column 
                    data <- myY_bayes %>% select(matches("Taxa") | matches("PSP3B.bayes.Protein.MC")) #select column that have trait in their name
                    print(data[1:2,])
                    print(sum(is.na(data)))  # Count missing values
                    #GAPIT(Y=data, G=myG, PCA.total = 6, model=c("GLM", "MLM", "CMLM", "ECMLM", "SUPER", "MLMM", "FarmCPU", "Blink"), Model.selection = TRUE)
                    GAPIT(Y=data, G=myG, PCA.total = 10, model=c("GLM", "MLMM", "FarmCPU", "Blink"), Model.selection = TRUE) #GOOD
                    #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("MLM"), Model.selection = TRUE) #Error in XCVI %*% beta.I : non-conformable arguments)
                    #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("CMLM"), Model.selection = TRUE) #Error in XCVI %*% beta.I : non-conformable arguments)
                    #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("ECMLM"), Model.selection = TRUE) #Error in XCVI %*% beta.I : non-conformable arguments)
                    #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("SUPER"), Model.selection = TRUE) #Error in GAPIT.Main(Y = ic_Y, GD = IC$GD[, -1], allGD = IC$myallGD[, -1],  :object 'KI.save' not found
                    #GAPIT(Y=data, G=myG, PCA.total = 10, model=c("FarmCPU"), Model.selection = TRUE) 
                    print("#############################################")
                  }
                  warnings()

print("#-------------------------------Run GAPIT with BAYES, for model selection for all envs except CLEM22-------------------------------#")

traits <- c("Moisture", "Protein.MC", "SAA.MC", "Fiber.MC", "T_starch.MC", "PDg")
traits
     
for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  #setwd
  setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/model_selection/")
  print(getwd())
  #select trait column 
  data <- myY_bayes %>% select(matches("Taxa") | matches(i) & !matches("CLEM22")) #select column that have trait in their name
  print(data[1:2,])
  GAPIT(Y=data, G=myG,
        PCA.total = 10,
        model=c("GLM", "MLMM", "FarmCPU", "Blink"),
        Model.selection = TRUE,
        kinship.algorithm = "VanRaden")
  print("#############################################")
}
warnings()

print("#-------------------------------Run GAPIT with BAYES, for model selection only for CLEM22-------------------------------#")

traits <- c("Moisture", "Protein.MC", "Fiber.MC", "T_starch.MC", "PDg")
traits
     
for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  #setwd
  setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/model_selection/")
  print(getwd())
  #select trait column 
  data <- myY_bayes %>% select(matches("Taxa") | matches(i) & matches("CLEM22")) #select column that have trait in their name
  print(data[1:2,])
  GAPIT(Y=data, G=myG,
        PCA.total = 10,
        model=c("GLM", "MLMM", "FarmCPU", "Blink"),
        Model.selection = TRUE,
        kinship.algorithm = "VanRaden")
  print("#############################################")
}
warnings()

print("#-------------------------------Check optimal PC number-------------------------------#")

traits <- c("Moisture", "Protein.MC", "SAA.MC", "Fiber.MC", "T_starch.MC", "PDg")
traits

#model selection files
files <- list.files('/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/model_selection', pattern = "BIC.Model.Selection.Results.csv", full.names=TRUE)
files

#check files
for (i in traits){
    print("#############################################")
    print(paste0("Reading model selection files for trait: ", i))
    files <- list.files('/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/model_selection/', pattern = paste0(i, ".BIC.Model.Selection.Results.csv"), full.names=TRUE)
    #print(file)
    for (f in files){
        print(paste0("Opening files: ", f))
        data <- read.csv(f) #read each file form 1 -10
        #print(head(data))
        BIC <- data %>% slice_max(BIC..larger.is.better....Schwarz.1978) %>% slice(1) #get row with max value in second column - BIC
        print(BIC)
    }
    print(paste0("END!!!!!: ", i))
    print("#############################################")
  }

```

#---> It seems like 0 PC is the best for: 
      - Moisture
      - Protein.MC
      - SAA.MC
      - Fiber.MC
      - T_starch.MC
      - PDg
      
#---> It seems like 1 PC is the best for: 
      - Protein.MC
      
#---> It seems like 2 PC is the best for: 
      - SAA.MC

PC with the highest BIC, means it offers the best balance between model fit and avoiding overfitting.

If interpretability or avoiding noise is a concern, sticking with PC with highes BIC is the best choice.
If maximizing fit (without worrying about overfitting) is your priority, then 2 or 3 PCs might also be considered.

###############################################
Removing the following model as causing issues:

ECMLM

Error in if (!DP$kinship.algorithm %in% c("FarmCPU", "MLMM", "BLINK",  : 
  argument is of length zero

```{r, echo=TRUE}
###############################################################################################################
```

# run GWAS - 0PC - all envs except 2022

______________________ Run GWAS in palmetto 
sbatch GAPIT_run_0PC.sbatch
squeue --me
                      
                      ______________________ GAPIT_run_0PC.sbatch
                      #!/bin/bash
                                            
                      #SBATCH --job-name GAPIT_run_0PC
                      #SBATCH --nodes 1
                      #SBATCH --tasks-per-node 1
                      #SBATCH --cpus-per-task 16
                      #SBATCH --mem 500gb
                      #SBATCH --time 72:00:00
                      #SBATCH --mail-type BEGIN
                      #SBATCH --mail-type END
                      #SBATCH --mail-type FAIL
                      #SBATCH --output=GAPIT_run_0PC.out
                      #SBATCH --error=GAPIT_run_0PC.err
                                            
                      cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS
                                            
                      module load r/4.4.1
                      Rscript GAPIT_run_0PC.R
                      ______________________
______________________

```{r, echo=FALSE}

print("#---------------------------------------------------------------------------------------------------------------------------------------#")
print("#-------------------------------------------View data BAYES----------------------------------#")

#Check data
myY_bayes[1:2,]
myG[1:5,1:12]

print("#---------------------------------------------------------------------------------------------------------------------------------------#")
print("#---------------------Run GAPIT with zero 0 PC for all envs except CLEM22--------------------#")

#make variables
traits <- c("Moisture", "Protein.MC", "SAA.MC", "Fiber.MC", "T_starch.MC", "PDg")
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  setwd(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".0PC", sep=""))
  print(getwd())
  #select trait column 
  data <- myY_bayes %>% dplyr::select(matches("Taxa") | matches(i) & !matches("CLEM22"))
  print(data[1:2,])
  GAPIT(Y=data, G=myG,
        PCA.total = 0,
        model=c("GLM", "MLM", "CMLM", "SUPER", "MLMM", "FarmCPU", "Blink"),
        Multiple_analysis = TRUE,
        kinship.algorithm = "VanRaden")
  print("#############################################")
}
warnings()

```

# run GWAS - 0PC - only 2022

______________________ Run GWAS in palmetto 
sbatch GAPIT_run_0PC_22.sbatch
squeue --me
                      
                      ______________________ GAPIT_run_0PC_22.sbatch
                      #!/bin/bash
                                            
                      #SBATCH --job-name GAPIT_run_0PC_22
                      #SBATCH --nodes 1
                      #SBATCH --tasks-per-node 1
                      #SBATCH --cpus-per-task 16
                      #SBATCH --mem 500gb
                      #SBATCH --time 72:00:00
                      #SBATCH --mail-type BEGIN
                      #SBATCH --mail-type END
                      #SBATCH --mail-type FAIL
                      #SBATCH --output=GAPIT_run_0PC_22.out
                      #SBATCH --error=GAPIT_run_0PC_22.err
                                            
                      cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS
                                            
                      module load r/4.4.1
                      Rscript GAPIT_run_0PC_22.R
                      ______________________
______________________

```{r, echo=FALSE}

print("#---------------------------------------------------------------------------------------------------------------------------------------#")
print("#-------------------------------------------View data BAYES----------------------------------#")

#Check data
myY_bayes[1:2,]
myG[1:5,1:12]

print("#---------------------------------------------------------------------------------------------------------------------------------------#")
print("#----------------------Run GAPIT with zero 0 PC only for CLEM22--------------------#")

traits <- c("Moisture", "Protein.MC", "Fiber.MC", "T_starch.MC", "PDg")
traits
     
for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  setwd(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".0PC_22", sep=""))
  print(getwd())
  #select trait column 
  data <- myY_bayes %>% dplyr::select(matches("Taxa") | matches(i) & matches("CLEM22")) #select column that have trait in their name
  print(data[1:2,])
  GAPIT(Y=data, G=myG,
        PCA.total = 0,
        model=c("GLM", "MLM", "CMLM", "MLMM", "FarmCPU", "Blink"), #remove "SUPER" as causing issues, Calls: GAPIT ... GAPIT.Main -> GAPIT.Bread -> GAPIT.SUPER.FastMLM -> diag
        Multiple_analysis = TRUE,
        kinship.algorithm = "VanRaden")
  print("#############################################")
}
warnings()

```

# run GWAS - 1PC and 2PC

______________________ Run GWAS in palmetto 
sbatch GAPIT_run_1PC_2PC.sbatch
squeue --me
                      
                      ______________________ GAPIT_run_1PC_2PC.sbatch
                      #!/bin/bash
                                            
                      #SBATCH --job-name GAPIT_run_1PC_2PC
                      #SBATCH --nodes 1
                      #SBATCH --tasks-per-node 1
                      #SBATCH --cpus-per-task 16
                      #SBATCH --mem 500gb
                      #SBATCH --time 72:00:00
                      #SBATCH --mail-type BEGIN
                      #SBATCH --mail-type END
                      #SBATCH --mail-type FAIL
                      #SBATCH --output=GAPIT_run_1PC_2PC.out
                      #SBATCH --error=GAPIT_run_1PC_2PC.err
                                            
                      cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS
                                            
                      module load r/4.4.1
                      Rscript GAPIT_run_1PC_2PC.R
                      ______________________
______________________

```{r, echo=FALSE}

print("#---------------------------------------------------------------------------------------------------------------------------------------#")
print("#-------------------------------Run GAPIT with 1 PC-------------------------------#")

#make variables
traits <- c("Protein.MC")
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  setwd(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".1PC", sep=""))
  print(getwd())
  #select trait column 
  data <- myY_bayes %>% dplyr::select(matches("Taxa") | matches(i))
  print(data[1:2,])
  GAPIT(Y=data, G=myG,
        PCA.total = 0,
        model=c("GLM", "MLM", "CMLM", "MLMM", "FarmCPU", "Blink"), #remove "SUPER" as causing issues, Calls: GAPIT ... GAPIT.Main -> GAPIT.Bread -> GAPIT.SUPER.FastMLM -> diag
        Multiple_analysis = TRUE,
        kinship.algorithm = "VanRaden")
  print("#############################################")
}
warnings()


print("#---------------------------------------------------------------------------------------------------------------------------------------#")
print("#-------------------------------Run GAPIT with BAYES 2 PC-------------------------------#")

#make variables
traits <- c("SAA.MC")
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  setwd(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".2PC", sep=""))
  print(getwd())
  #select trait column 
  data <- myY_bayes %>% dplyr::select(matches("Taxa") | matches(i) & !matches("CLEM22"))
  print(data[1:2,])
  GAPIT(Y=data, G=myG,
        PCA.total = 0,
        model=c("GLM", "MLM", "CMLM", "SUPER", "MLMM", "FarmCPU", "Blink"),
        Multiple_analysis = TRUE,
        kinship.algorithm = "VanRaden")
  print("#############################################")
}
warnings()

```

# ############################################################################################
# ANALIZE RESULTS

folders in /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE

Moisture.0PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/Moisture.0PC

Protein.MC.0PC and Protein.MC.1PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/Protein.MC.0PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/Protein.MC.1PC

SAA.MC.0PC and SAA.MC.2PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/SAA.MC.0PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/SAA.MC.2PC

Fiber.MC.0PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/Fiber.MC.0PC

T_starch.MC.0PC 
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/T_starch.MC.0PC

PDg.0PC
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/PDg.0PC


file to look for: 

- GAPIT.Association.GWAS_Results.{MODEL/TRAIT} --> SNP,Chr,Pos,P.value,MAF,nobs,H&B.P.Value,Effect
- GAPIT.Association.GWAS_StdErr.{MODEL/TRAIT}  --> SNP,Chromosome,Position,DF,t Value,std Error,effect
- GAPIT.Association.PVE.{MODEL/TRAIT}          --> SNP,Chromosome,Position,P.value,maf,nobs,effect,Phenotype_Variance_Explained(%)

- GAPIT.Association.Filter_GWAS_results.csv    --> ,SNP,Chr,Pos,P.value,MAF,traits (1 file per run)

# #################################################################################
#### Put GWAS together using all GAPIT.Association.GWAS_Results.{MODEL/TRAIT} files

Make Manhattan plots and qqplots

```{r, include=TRUE}

#save figures
setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/figures")
getwd()

models <- c("GLM", "MLM", "CMLM", "SUPER", "MLMM", "FarmCPU", "BLINK")
#models <- c("BLINK")
models
models_22 <- c("GLM", "MLM", "CMLM", "MLMM", "FarmCPU", "BLINK")
#models_22 <- c("BLINK")
models_22

envs_mod <- c("PSP3B", "PSP3NB", "RA20CL21B", "RA20CL21NB", "RAWL20B", "RAWL20NB", "CLEM21B", "CLEM21NB")
#envs_mod <- c("PSP3B")
envs_mod
envs_mod_22 <- c("CLEM22B", "CLEM22NB")
#envs_mod_22 <- c("CLEM22B")
envs_mod_22

traits <- c("Moisture", "Protein.MC", "SAA.MC", "Fiber.MC", "T_starch.MC", "PDg")
#traits <- c("Protein.MC")
traits
traits_22 <- c("Moisture", "Protein.MC", "Fiber.MC", "T_starch.MC", "PDg")
#traits_22 <- c("Protein.MC")
traits_22

#get column for merging, SNP, chr, and pos
df <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/Moisture.0PC/GAPIT.Association.GWAS_Results.GLM.PSP3NB.bayes.Moisture.csv") #could be any file, just need 3 columns
markers_P <- unique(df[,c(1,2,3),drop=F])
head(markers_P)

#read all file sand edit column names and all, and merge: All env except 2022
for (m in models){
  for (em in envs_mod){
    for (t in traits){
      print("#############################################")
      print(paste0("Processing: ", m, ".", em, ".bayes.", t, sep="."))
      print(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/", t, ".0PC", "/GAPIT.Association.GWAS_Results.", m, ".", em, ".bayes.", t, ".csv"))
      gwas <- read.csv(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/", t, ".0PC", "/GAPIT.Association.GWAS_Results.", m, ".", em, ".bayes.", t, ".csv"), head = TRUE)
      colnames(gwas)[4:8] <- sapply(colnames(gwas)[4:8], function(x) paste(x, ".", m, ".", em,".", t, sep ="")) #add info (trait, env, model) to columns\
      gwas <-  gwas %>% dplyr::select(1,2,3, matches("P.value."))
      #print(head(gwas))
      markers_P <- merge(markers_P, gwas, all = T)
      #print(head(markers_P))
      print("#############################################")
    }
  }
}
#read all file sand edit column names and all, and merge: ONLY 2022
for (m in models_22){
  for (em in envs_mod_22){
    for (t in traits_22){
      print("#############################################")
      print(paste0("Processing: ", m, ".", em, ".bayes.", t, sep="."))
      print(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/", t, ".0PC_22", "/GAPIT.Association.GWAS_Results.", m, ".", em, ".bayes.", t, ".csv"))
      gwas <- read.csv(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/", t, ".0PC_22", "/GAPIT.Association.GWAS_Results.", m, ".", em, ".bayes.", t, ".csv"), head = TRUE)
      colnames(gwas)[4:8] <- sapply(colnames(gwas)[4:8], function(x) paste(x, ".", m, ".", em,".", t, sep ="")) #add info (trait, env, model) to columns\
      gwas <-  gwas %>% dplyr::select(1,2,3, matches("P.value."))
      #print(head(gwas))
      markers_P <- merge(markers_P, gwas, all = T)
      #print(head(markers_P))
      print("#############################################")
    }
  }
}

print(head(markers_P))
#rename chromosomes and columns
markers_Pval <- markers_P %>% dplyr::mutate(Chr = str_replace(Chr, "1LG6", "1")) %>% mutate(Chr = str_replace(Chr, "2LG1", "2")) %>% 
                                             mutate(Chr = str_replace(Chr, "3LG5", "3")) %>% mutate(Chr = str_replace(Chr, "4LG4", "4")) %>% 
                                             mutate(Chr = str_replace(Chr, "5LG3", "5")) %>% mutate(Chr = str_replace(Chr, "6LG2", "6")) %>% 
                                             mutate(Chr = str_replace(Chr, "7LG7", "7")) %>% rename("Chromosome" = "Chr") %>% rename("Position" = "Pos")
markers_Pval[1:2,]
str(markers_Pval)
markers_Pval$Chromosome <- as.numeric(markers_Pval$Chromosome) #chromosome as numeric
str(markers_Pval)
markers_Pval[1:2,]
dim(markers_Pval)

          # example of what I need to do
          #select column for plots
          manhaPvalue <-  markers_Pval %>% dplyr::select(1,2,3, matches("P.value.") & matches("Protein") & matches("BLINK") & !matches("H.B.P.Value.") & !matches("nobs") & !matches("MAF") & !matches("Effect"))
          manhaPvalue[1:2,]
          # manhaHBPValue  <-  markers_Pval %>% dplyr::select(1,2,3, matches("H.B.P.Value.") & matches("Protein") & matches("BLINK") & !matches("nobs") & !matches("MAF") & !matches("Effect"))
          # manhaHBPValue[1:2,]
          
          threshold5=0.05/nrow(manhaPvalue)
          threshold5
          -log10(threshold5)
          threshold1=0.01/nrow(manhaPvalue)
          threshold1
          -log10(threshold1)
          
          #pvalue
          CMplot(manhaPvalue, plot.type="m",multraits=TRUE,threshold=c(threshold5,threshold1),threshold.lty=c(1,2), 
                  threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
                  signal.cex=1, file="jpg",file.name="Protein_blink",dpi=300,file.output=FALSE,verbose=TRUE,
                  points.alpha=100,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=6,width=11, cex=0.3,
                  main="Protein - BLINK", main.cex=1.2)
          CMplot(manhaPvalue, plot.type="q",multraits=TRUE,threshold=c(threshold5),threshold.lty=c(1,2), 
                  threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
                  signal.cex=1, file="jpg",file.name="Protein_blink",dpi=300,file.output=FALSE,verbose=TRUE,
                  points.alpha=200,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=6,width=6, cex=0.3, main.cex=1.2,
                  conf.int=TRUE)

#columns I need to remove, because GWAS does not look good
markers_Pval[1:2,]
markers_Pval <- markers_Pval %>% select(-c(P.value.GLM.CLEM22B.Protein.MC,
                                           P.value.GLM.CLEM22NB.Protein.MC,
                                           P.value.MLM.CLEM22B.Protein.MC,
                                           P.value.MLM.CLEM22NB.Protein.MC,
                                           P.value.CMLM.CLEM22B.Protein.MC,
                                           P.value.CMLM.CLEM22NB.Protein.MC,
                                           P.value.MLMM.CLEM22B.Protein.MC,
                                           P.value.FarmCPU.CLEM22B.Protein.MC,
                                           P.value.GLM.RAWL20B.SAA.MC,
                                           P.value.GLM.RAWL20NB.SAA.MC,
                                           P.value.SUPER.RAWL20B.SAA.MC,
                                           P.value.SUPER.RAWL20NB.SAA.MC,
                                           P.value.MLMM.RA20CL21B.SAA.MC,
                                           P.value.MLMM.PSP3B.SAA.MC,
                                           P.value.FarmCPU.CLEM22B.PDg,
                                           P.value.BLINK.CLEM22NB.PDg))        
#loop to create the plots
models <- c("GLM", "CMLM", "SUPER", "MLMM", "FarmCPU", "BLINK")
models
traits <- c("Moisture", "Protein", "SAA", "Fiber", "T_starch", "PDg")
traits

for (m in models){
    for (t in traits){
      print("#############################################")
      print(paste0("Processing: ", t, " - ", m))
      #print(markers_Pval[1:2,])
      manhaPvalue <-  markers_Pval %>% dplyr::select(1,2,3, matches("P.value.") & matches(t) & matches(m) & !matches("H.B.P.Value.") & !matches("nobs") & !matches("MAF") & !matches("Effect"))
      print(manhaPvalue[1:2,])
      print(dim(manhaPvalue))
      threshold5=0.05/nrow(manhaPvalue)
      print(threshold5)
      threshold1=0.01/nrow(manhaPvalue)
      print(threshold1)
      CMplot(manhaPvalue, plot.type="m",multraits=TRUE,threshold=c(threshold5,threshold1),threshold.lty=c(1,2),
        threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
        signal.cex=1, file="jpg",file.name=paste0(t,"_",m),dpi=300,file.output=TRUE,verbose=TRUE,
        points.alpha=100,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=5,width=9, cex=0.3,
        main=paste0(t," - ",m), main.cex=1.2)
      CMplot(manhaPvalue, plot.type="q",multraits=TRUE,threshold=threshold5,
        threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
        signal.cex=1, file="jpg",file.name=paste0(t,"_",m),dpi=300,file.output=TRUE,verbose=TRUE,
        points.alpha=200,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=5,width=9, cex=0.3, main.cex=1.2,
        conf.int=TRUE)
      print("#############################################")
    }
}


#loop to create the plots, this is just for MLM, because I need to remove CMLM and MLMM
models <- c("MLM")
models
traits <- c("Moisture", "Protein", "SAA", "Fiber", "T_starch", "PDg")
traits

for (m in models){
    for (t in traits){
      print("#############################################")
      print(paste0("Processing: ", t, " - ", m))
      #print(markers_Pval[1:2,])
      manhaPvalue <-  markers_Pval %>% dplyr::select(1,2,3, matches("P.value.") & matches(t) & matches(m) & !matches("H.B.P.Value.") & 
                                                       !matches("nobs") & !matches("MAF") & !matches("Effect") & !matches("CMLM") & !matches("MLMM"))
      print(manhaPvalue[1:2,])
      print(dim(manhaPvalue))
      threshold5=0.05/nrow(manhaPvalue)
      print(threshold5)
      threshold1=0.01/nrow(manhaPvalue)
      print(threshold1)
      CMplot(manhaPvalue, plot.type="m",multraits=TRUE,threshold=c(threshold5,threshold1),threshold.lty=c(1,2),
        threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
        signal.cex=1, file="jpg",file.name=paste0(t,"_",m),dpi=300,file.output=TRUE,verbose=TRUE,
        points.alpha=100,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=5,width=9, cex=0.3,
        main=paste0(t," - ",m), main.cex=1.2)
      CMplot(manhaPvalue, plot.type="q",multraits=TRUE,threshold=threshold5,
        threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
        signal.cex=1, file="jpg",file.name=paste0(t,"_",m),dpi=300,file.output=TRUE,verbose=TRUE,
        points.alpha=200,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=5,width=9, cex=0.3, main.cex=1.2,
        conf.int=TRUE)
      print("#############################################")
    }
}

```

# ############################################################################################
#### Put significant SNPs together using all GAPIT.Association.PVE.{MODEL/TRAIT}  files

read files with significant SNPs and put all together

```{r, include=TRUE}

#WD
setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps")
getwd()

# Load sample file to copy structure (correct types)
sample_file <- "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/Protein.MC.0PC/GAPIT.Association.PVE.GLM.RA20CL21NB.bayes.Protein.MC.csv"
sample_data <- read.csv(sample_file, stringsAsFactors = FALSE)
sample_data
# Rename column if needed
sample_data <- sample_data %>% dplyr::rename("Variance_Explained" = matches("Phenotype_Variance_Explained"))

# Add extra column for filename
sample_data$filename <- as.character(NA)

# Create empty dataframe with correct structure
allsigsnps <- sample_data[0, ]  # ← this is key, select only column names
str(allsigsnps)
head(allsigsnps)

print("#-------------------------------read sign SNPs with 0 PC-------------------------------#")
#make variables, #run for 0 PCA
traits <- c("Moisture", "Protein.MC", "SAA.MC", "Fiber.MC", "T_starch.MC", "PDg")
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  print(paste0("Processing trait: 0PC results"))
  filenames <- list.files(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".0PC", sep=""), pattern = "GAPIT.Association.PVE", full.names=TRUE)
  print(filenames)
  
        for(f in 1:length(filenames)){
          print(paste0("Processing file: ", filenames[f]))
          data <- read.csv(filenames[f], head = TRUE) #open each file
          data <- data %>% dplyr::rename("Variance_Explained" = "Phenotype_Variance_Explained...")
          colnames(data)[2] <- "Chromosome"
          colnames(data)[3] <- "Position"
          
            # Force column types (e.g., SNP as character)
            if ("SNP" %in% colnames(data)) {
              data$SNP <- as.character(data$SNP)
            }

          data$filename <- basename(filenames[f]) #add origin of file, path and name file
          print(head(data)) #check data
          print(str(data)) #check data
          data$PC_Number <- 0
          allsigsnps <- bind_rows(allsigsnps,data)
        }
  print("#############################################")
}
allsigsnps
head(allsigsnps)
tail(allsigsnps)
nrow(allsigsnps)

print("#-------------------------------read sign SNPs with 0 PC for 2022-------------------------------#")
#make variables, #run for 0 PCA
traits <- c("Moisture", "Protein.MC", "PDg") #no significant SNPs for Fiber.MC, T_starch.MC
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  print(paste0("Processing trait: 0PC_22 results"))
  filenames <- list.files(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".0PC_22", sep=""), pattern = "GAPIT.Association.PVE", full.names=TRUE)
  print(filenames)
  
        for(f in 1:length(filenames)){
          print(paste0("Processing file: ", filenames[f]))
          data <- read.csv(filenames[f], head = TRUE) #open each file
          data <- data %>% dplyr::rename("Variance_Explained" = "Phenotype_Variance_Explained...")
          colnames(data)[2] <- "Chromosome"
          colnames(data)[3] <- "Position"
          
            # Force column types (e.g., SNP as character)
            if ("SNP" %in% colnames(data)) {
              data$SNP <- as.character(data$SNP)
            }

          data$filename <- basename(filenames[f]) #add origin of file, path and name file
          print(head(data)) #check data
          print(str(data)) #check data
          data$PC_Number <- 0
          allsigsnps <- bind_rows(allsigsnps,data)
        }
  print("#############################################")
}
allsigsnps
head(allsigsnps)
tail(allsigsnps)
nrow(allsigsnps)

print("#-------------------------------read sign SNPs with 1 PC-------------------------------#")
#make variables, #run for 1 PCA
traits <- c("Protein.MC")
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  print(paste0("Processing trait: 1PC results"))
  filenames <- list.files(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".1PC", sep=""), pattern = "GAPIT.Association.PVE", full.names=TRUE)
  print(filenames)
  
        for(f in 1:length(filenames)){
          print(paste0("Processing file: ", filenames[f]))
          data <- read.csv(filenames[f], head = TRUE) #open each file
          data <- data %>% dplyr::rename("Variance_Explained" = "Phenotype_Variance_Explained...")
          colnames(data)[2] <- "Chromosome"
          colnames(data)[3] <- "Position"
          
            # Force column types (e.g., SNP as character)
            if ("SNP" %in% colnames(data)) {
              data$SNP <- as.character(data$SNP)
            }

          data$filename <- basename(filenames[f]) #add origin of file, path and name file
          print(head(data)) #check data
          print(str(data)) #check data
          data$PC_Number <- 1
          allsigsnps <- bind_rows(allsigsnps,data)
        }
  print("#############################################")
}
allsigsnps
head(allsigsnps)
tail(allsigsnps)
nrow(allsigsnps)

print("#-------------------------------read sign SNPs with 2 PC-------------------------------#")
#make variables, #run for 3 PCA
traits <- c("SAA.MC")
traits

for (i in traits){
  print("#############################################")
  print(paste0("Processing trait: ", i))
  print(paste0("Processing trait: 2PC results"))
  filenames <- list.files(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/",i, ".2PC", sep=""), pattern = "GAPIT.Association.PVE", full.names=TRUE)
  print(filenames)
  
        for(f in 1:length(filenames)){
          print(paste0("Processing file: ", filenames[f]))
          data <- read.csv(filenames[f], head = TRUE) #open each file
          data <- data %>% dplyr::rename("Variance_Explained" = "Phenotype_Variance_Explained...")
          colnames(data)[2] <- "Chromosome"
          colnames(data)[3] <- "Position"
          
            # Force column types (e.g., SNP as character)
            if ("SNP" %in% colnames(data)) {
              data$SNP <- as.character(data$SNP)
            }

          data$filename <- basename(filenames[f]) #add origin of file, path and name file
          print(head(data)) #check data
          print(str(data)) #check data
          data$PC_Number <- 2
          allsigsnps <- bind_rows(allsigsnps,data)
        }
  print("#############################################")
}
allsigsnps
head(allsigsnps)
tail(allsigsnps)
nrow(allsigsnps)

print("#-------------------------------merge nad save files-------------------------------#")

#write.csv(allsigsnps,"allsigsnps.csv", row.names=FALSE)

#copy file name to be splitted
allsigsnps$filenameSPLIT <- allsigsnps$filename 
head(allsigsnps)
#need to chaqnce T_starch by Tstarch, do when spliting filename does nor cause issues
allsigsnps <- allsigsnps %>% mutate(filenameSPLIT = str_replace(filenameSPLIT, "T_starch", "Tstarch"))
#split file name 
allsigsnps2 <- separate(allsigsnps, 
                        col = filenameSPLIT,  # This should be the correct column name
                        into = c("GAPIT", "Association", "PVE", "Model", "Env", "bayes", "Trait", "csv"),
                        extra = "merge",   # This will merge the extra pieces into the last column
                        fill = "right")    # Fill missing values with NAs
#select columns to save
allsigsnps2 <- allsigsnps2 %>% select(c(1:6,9,10,19,16,17,12,11)) %>% arrange(Chromosome, Position)
head(allsigsnps2)
tail(allsigsnps2)
str(allsigsnps2)

#check and SNPs that I removed from Manhattan
allsigsnps2 <- allsigsnps2 %>% filter(!str_detect(filename, "GLM.CLEM22B.bayes.Protein") &
                       !str_detect(filename, "GLM.CLEM22NB.bayes.Protein") &
                       !str_detect(filename, "MLM.CLEM22B.bayes.Protein") &
                       !str_detect(filename, "MLM.CLEM22NB.bayes.Protein") &
                       !str_detect(filename, "CMLM.CLEM22B.bayes.Protein") &
                       !str_detect(filename, "CMLM.CLEM22NB.bayes.Protein") &
                       !str_detect(filename, "MLMM.CLEM22B.bayes.Protein") &
                       !str_detect(filename, "FarmCPU.CLEM22B.bayes.Protein") &
                       !str_detect(filename, "GLM.RAWL20B.bayes.SAA") &
                       !str_detect(filename, "GLM.RAWL20NB.bayes.SAA") &
                       !str_detect(filename, "SUPER.RAWL20B.bayes.SAA") &
                       !str_detect(filename, "SUPER.RAWL20NB.bayes.SAA") &
                       !str_detect(filename, "MLMM.RA20CL21B.bayes.SAA") &
                       !str_detect(filename, "MLMM.PSP3B.bayes.SAA") &
                       !str_detect(filename, "FarmCPU.CLEM22B.bayes.PDg") &
                       !str_detect(filename, "BLINK.CLEM22NB.bayes.PDg")) 

#make sure those are out
str(allsigsnps2 %>% filter(str_detect(filename, "GLM.CLEM22B.bayes.Protein") |
                       str_detect(filename, "GLM.CLEM22NB.bayes.Protein") |
                       str_detect(filename, "MLM.CLEM22B.bayes.Protein") |
                       str_detect(filename, "MLM.CLEM22NB.bayes.Protein") |
                       str_detect(filename, "CMLM.CLEM22B.bayes.Protein") |
                       str_detect(filename, "CMLM.CLEM22NB.bayes.Protein") |
                       str_detect(filename, "MLMM.CLEM22B.bayes.Protein") |
                       str_detect(filename, "FarmCPU.CLEM22B.bayes.Protein") |
                       str_detect(filename, "GLM.RAWL20B.bayes.SAA") |
                       str_detect(filename, "GLM.RAWL20NB.bayes.SAA") |
                       str_detect(filename, "SUPER.RAWL20B.bayes.SAA") |
                       str_detect(filename, "SUPER.RAWL20NB.bayes.SAA") |
                       str_detect(filename, "MLMM.RA20CL21B.bayes.SAA") |
                       str_detect(filename, "MLMM.PSP3B.bayes.SAA") |
                       str_detect(filename, "FarmCPU.CLEM22B.bayes.PDg") |
                       str_detect(filename, "BLINK.CLEM22NB.bayes.PDg")))  

#make summary, and colaps by uniqeu SNPs
as.data.frame(allsigsnps2 %>% dplyr::group_by(SNP, Chromosome, Position) %>% dplyr::arrange(Chromosome, Position) %>% dplyr::summarize(countTotal = n()))

print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")
unique_sigsnps_summary <- as.data.frame(allsigsnps2 %>% dplyr::group_by(SNP, Chromosome, Position) %>% 
                                                                                dplyr::summarize(countTotal = n(), 
                                                                                Traits = paste(unique(Trait), collapse = ", "), 
                                                                                Models = paste(unique(Model), collapse = ", "),
                                                                                Envs = paste(unique(Env), collapse = ", "), 
                                                                                PCs = paste(unique(PC_Number), collapse = ", "),
                                                                                mean_P.value = paste(round(mean(P.value), 10), collapse = ", "),
                                                                                log10mean_P.value = paste(round(mean(-log10(P.value)), 3), collapse = ", "),
                                                                                mean_MAF = round(mean(maf), digits = 3),
                                                                                mean_nobs = round(mean(nobs), digits = 1),
                                                                                mean_effect = round(mean(effect), digits = 3),
                                                                                mean_VE = mean(Variance_Explained),
                                                                                .groups = "drop")) %>% arrange(Chromosome, Position)
head(unique_sigsnps_summary)

print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")

# threshold5 <- 9.20556e-07
# threshold1 <- 1.841112e-07

threshold5=0.05/54315
threshold5
-log10(threshold5)
threshold1=0.01/54315
threshold1
-log10(threshold1)

#manhattan all chrs
unique_sigsnps_summary <- unique_sigsnps_summary %>% filter(mean_MAF >= 0.05 & mean_VE >= 0.1)
unique_sigsnps_summary_bychr <- unique_sigsnps_summary
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="AllChrs",dpi=300,file.output=TRUE,verbose=TRUE,width=18,height=4.5,cex=0.3, main="Significant SNPs", main.cex=1.2)

SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="AllChrs_models",dpi=300,file.output=TRUE,verbose=TRUE,width=18,height=4.5,cex=0.3, main="Significant SNPs", main.cex=1.2)

#manhattan by chr1
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 1)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr1",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr1", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 1)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr1_models",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr1", main.cex=1.2)

#manhattan by chr2
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 2)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr2",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr2", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 2)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr2_models",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr2", main.cex=1.2)

#manhattan by chr3
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 3)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr3",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr3", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 3)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr3_models",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr3", main.cex=1.2)

#manhattan by chr4
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 4)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr4",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr4", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 4)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr4_models",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr4", main.cex=1.2)

#manhattan by chr5
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 5)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr5",dpi=300,file.output=TRUE,verbose=TRUE,width=18,height=4.5,cex=0.3, main="Significant SNPs in Chr5", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 5)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr5_models",dpi=300,file.output=TRUE,verbose=TRUE,width=18,height=4.5,cex=0.3, main="Significant SNPs in Chr5", main.cex=1.2)

#manhattan by chr6
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 6)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr6",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr6", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 6)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr6_models",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr6", main.cex=1.2)

#manhattan by chr7
unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 7)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- unique_sigsnps_summary_bychr$Traits
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr7",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr7", main.cex=1.2)

unique_sigsnps_summary_bychr <- unique_sigsnps_summary %>% filter(Chromosome == 7)
unique_sigsnps_summary_bychr
SNPs <- unique_sigsnps_summary_bychr$SNP
SNPs
genes <- paste0(unique_sigsnps_summary_bychr$Traits," (",unique_sigsnps_summary_bychr$Models,")")
genes
set.seed(1234)
CMplot((unique_sigsnps_summary_bychr %>% select(c(1:3,9))), plot.type="m",LOG10=TRUE,col=c("grey30","grey60"),highlight=SNPs,
        highlight.col=rep(c("#377eb8"),length=length(SNPs)),highlight.cex=1, highlight.text=genes,      
        highlight.text.col=rep("black",length(SNPs)),threshold=c(threshold5,threshold1),threshold.lty=c(1,2), threshold.col=c("darkslategray4","#4daf4a"),
        amplify=FALSE,file="jpg",file.name="Chr7_models",dpi=300,file.output=TRUE,verbose=TRUE,width=9,height=4.5,cex=0.3, main="Significant SNPs in Chr7", main.cex=1.2)

print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")
#ONYL significant SNPs as Manhattan
Man_unique_sigsnps_summary<- unique_sigsnps_summary %>% select(c(1,2,3,9))
head(Man_unique_sigsnps_summary)

threshold5 <- 9.20556e-07
threshold1 <- 1.841112e-07
          
#pvalue
CMplot(Man_unique_sigsnps_summary, plot.type="m",threshold=c(threshold5,threshold1),threshold.lty=c(1,2), 
       threshold.lwd=c(1,1), threshold.col=c("black","black"), amplify=TRUE,bin.size=1e6,
       signal.cex=1, file="jpg",file.name="Protein_blink",dpi=300,file.output=FALSE,verbose=TRUE,
       points.alpha=100,legend.ncol=3,legend.pos="middle",legend.cex=0.7, height=6,width=11, cex=0.3,
       main="SNPs above 5% and 1%", main.cex=1.2)

print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")
#same file for LDBlockShow
chr_site_Pvalue <- unique_sigsnps_summary %>% select(c(2,3,9)) %>% rename("chr" = "Chromosome", "site" = "Position", "Pvalue" = "mean_P.value") 
chr_site_Pvalue$Pvalue <- as.numeric(chr_site_Pvalue$Pvalue) #chromosome as numeric
head(chr_site_Pvalue)
str(chr_site_Pvalue)
print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")
#get unique SNPs, chromosome and position
unique_sigsnps_chr_pos <- unique(allsigsnps2[,c(1,2,3),drop=F])
unique_sigsnps_chr_pos <- unique_sigsnps_chr_pos %>% arrange(Chromosome, Position)
head(unique_sigsnps_chr_pos)
print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")
#get unique SNPs list, rename chromosomes to be compatible with plink
unique_sigsnps_list <- allsigsnps2 %>% dplyr::mutate(SNP = str_replace(SNP, "S1LG6_", "1:")) %>% mutate(SNP = str_replace(SNP, "S2LG1_", "2:")) %>% 
                                                      mutate(SNP = str_replace(SNP, "S3LG5_", "3:")) %>% mutate(SNP = str_replace(SNP, "S4LG4_", "4:")) %>% 
                                                      mutate(SNP = str_replace(SNP, "S5LG3_", "5:")) %>% mutate(SNP = str_replace(SNP, "S6LG2_", "6:")) %>% 
                                                      mutate(SNP = str_replace(SNP, "S7LG7_", "7:"))
unique_sigsnps_list2 <- unique(unique_sigsnps_list$SNP)
unique_sigsnps_list2
class(unique_sigsnps_list2)
length(unique_sigsnps_list2)
print("#-------------------------------------------------------------------------------------------------------------------------------------------------------------#")
# Export CSVs for record
write.csv(allsigsnps2, file = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/allsigsnps.csv", row.names = FALSE)
write.csv(unique_sigsnps_summary, file = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_summary.csv", row.names = FALSE)
write.csv(unique_sigsnps_chr_pos, file = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_chr_pos.csv", row.names = FALSE)
write.table(unique_sigsnps_list2, file = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_list.txt", row.names = FALSE, col.names = FALSE)
write.table(chr_site_Pvalue, file = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_chr_site_Pvalue.txt", row.names = FALSE, col.names = FALSE)

```

```{r, echo=TRUE}
###############################################################################################################
```

test significance of SIG SNPs

```{r, echo=TRUE}

#---------------------------------Read Genotype ---------------------------------------#
#267 accessions
geno <- fread('/project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.012')
geno[1:2,1:10]
## fread adds a column for rownumbers which is why I am removing the first column and then converting 0,1,2 format to -1,-0,1 genotype format
geno <- as.matrix(geno[,-1])
geno[1:2,1:10]
str(geno)
class(geno)

#Add genotype names
names <- fread('/project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.012.indv', header=F)
rownames(geno) <- names$V1
geno[1:2,1:10]
str(geno) #267 accessiona and 54315 Chromosomes
class(geno)

#Add marker names
markers <- fread('/project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.012.pos', header=F)
markers$position <- paste(markers$V1, markers$V2, sep="_")
colnames(geno) <- markers$position
geno[1:2,1:10]
str(geno)
class(geno)
dim(geno)

#Read Chromosome list
SNP_LIST_CP <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_chr_pos.csv", stringsAsFactors = FALSE)
str(SNP_LIST_CP)
head(SNP_LIST_CP)
SNP_LIST_CP <- SNP_LIST_CP %>% mutate(across('SNP', str_replace, 'S', 'chr')) #mutate to match names in geno

#get marker data of significant SNPs
colnames(geno) %in% SNP_LIST_CP$SNP
geno[,colnames(geno) %in% SNP_LIST_CP$SNP, drop = FALSE] #drop = FALSE: ensures that if only one column is returned, it remains a data frame (not a vector).
GENO_SIG_SNPs <- as.data.frame(geno[,colnames(geno) %in% SNP_LIST_CP$SNP, drop = FALSE])
GENO_SIG_SNPs$Taxa <- rownames(GENO_SIG_SNPs)
GENO_SIG_SNPs[1:2,]
str(GENO_SIG_SNPs)
class(GENO_SIG_SNPs)
dim(GENO_SIG_SNPs)
GENO_SIG_SNPs %>% filter(Taxa == "SRR5350189")

#phenotype
myY_bayes <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_Phenotypes/PSP3_BAYESREE.csv", head = TRUE)
myY_bayes <- as.data.frame(myY_bayes)
str(myY_bayes)
myY_bayes[1:2,]
dim(myY_bayes)
myY_bayes %>% filter(Taxa == "SRR5350189")

#merge
GENO_SIG_SNPsLM <- Reduce(function(...) merge(..., fix.by='Taxa', all.x=TRUE), list(GENO_SIG_SNPs, myY_bayes)) #Put data together
GENO_SIG_SNPsLM[1:2,]
GENO_SIG_SNPsLM %>% filter(Taxa == "SRR5350189")


summary(lm(PSP3B.bayes.SAA.MC ~chr2LG1_5715307, data=GENO_SIG_SNPsLM))
p1 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr2LG1_5715307), y = PSP3B.bayes.SAA.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr2LG1_5715307", y = "SAA") +
  theme_minimal()
p1

summary(lm(PSP3B.bayes.SAA.MC ~chr2LG1_292238595, data=GENO_SIG_SNPsLM))
p2 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr2LG1_292238595), y = PSP3B.bayes.SAA.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr2LG1_292238595", y = "SAA") +
  theme_minimal()
p2

summary(lm(PSP3B.bayes.Protein.MC ~chr3LG5_219840019, data=GENO_SIG_SNPsLM))
p3 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr3LG5_219840019), y = PSP3B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr3LG5_219840019", y = "Protein content") +
  theme_minimal()
p3

summary(lm(RA20CL21B.bayes.Protein.MC ~chr3LG5_434614474, data=GENO_SIG_SNPsLM))
p4 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr3LG5_434614474), y = RA20CL21B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr3LG5_434614474", y = "Protein") +
  theme_minimal()
p4

summary(lm(PSP3B.bayes.Protein.MC ~chr4LG4_76004069, data=GENO_SIG_SNPsLM)) ######??????
p5 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr4LG4_76004069), y = PSP3B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr4LG4_76004069", y = "Protein") +
  theme_minimal()
p5

summary(lm(RAWL20B.bayes.SAA.MC ~chr4LG4_299351672, data=GENO_SIG_SNPsLM))
p6 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr4LG4_299351672), y = RAWL20B.bayes.SAA.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr4LG4_299351672", y = "SAA") +
  theme_minimal()
p6

summary(lm(RA20CL21B.bayes.Protein.MC ~chr4LG4_416335390, data=GENO_SIG_SNPsLM))
p7 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr4LG4_416335390), y = RA20CL21B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr4LG4_416335390", y = "Protein") +
  theme_minimal()
p7

summary(lm(PSP3B.bayes.SAA.MC ~chr4LG4_432425908, data=GENO_SIG_SNPsLM))
p8 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr4LG4_432425908), y = PSP3B.bayes.SAA.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr4LG4_432425908", y = "SAA") +
  theme_minimal()
p8

summary(lm(PSP3B.bayes.Protein.MC ~chr5LG3_61677899, data=GENO_SIG_SNPsLM))
p9 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_61677899), y = PSP3B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_61677899", y = "Protein") +
  theme_minimal()
p9

summary(lm(RA20CL21B.bayes.Protein.MC ~chr5LG3_65187310, data=GENO_SIG_SNPsLM))
p10 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_65187310), y = RA20CL21B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_65187310", y = "Protein") +
  theme_minimal()
p10

summary(lm(PSP3B.bayes.SAA.MC ~chr5LG3_65228308, data=GENO_SIG_SNPsLM))
p11 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_65228308), y = PSP3B.bayes.SAA.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_65228308", y = "SAA") +
  theme_minimal()
p11

summary(lm(PSP3NB.bayes.Protein.MC ~chr5LG3_68426550, data=GENO_SIG_SNPsLM))
p12 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_68426550), y = PSP3NB.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_68426550", y = "Protein") +
  theme_minimal()
p12

summary(lm(PSP3NB.bayes.Protein.MC ~chr5LG3_75020967, data=GENO_SIG_SNPsLM))
p13 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_75020967), y = PSP3NB.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_75020967", y = "Protein") +
  theme_minimal()
p13

summary(lm(PSP3NB.bayes.Protein.MC ~chr5LG3_75020977, data=GENO_SIG_SNPsLM))
p14 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_75020977), y = PSP3NB.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_75020977", y = "Protein") +
  theme_minimal()
p14

summary(lm(PSP3B.bayes.SAA.MC ~chr5LG3_78533957, data=GENO_SIG_SNPsLM))
p15 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr5LG3_78533957), y = PSP3B.bayes.SAA.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr5LG3_78533957", y = "SAA") +
  theme_minimal()
p15

summary(lm(RAWL20B.bayes.PDg ~chr6LG2_102304910, data=GENO_SIG_SNPsLM))
p16 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr6LG2_102304910), y = RAWL20B.bayes.PDg)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr6LG2_102304910", y = "PDg") +
  theme_minimal()
p16

summary(lm(RA20CL21B.bayes.Protein.MC ~chr7LG7_47365879, data=GENO_SIG_SNPsLM))
p17 <- ggplot(GENO_SIG_SNPsLM, aes(x = as.factor(chr7LG7_47365879), y = RA20CL21B.bayes.Protein.MC)) +
  geom_boxplot(fill = "lightblue", color = "black", outlier.shape = NA) +  # hide default outliers
  geom_jitter(width = 0.2, size = 1.5, alpha = 0.6, color = "black") +  # add points
  labs(x = "chr7LG7_47365879", y = "Protein") +
  theme_minimal()
p17


# Combine plots using cowplot::plot_grid
plot_grid(p1, p2, p6, p8, p11, p15, labels = c("A", "B", "C", "D", "E", "F"), ncol = 6, align = "v")
plot_grid(p3, p4, p5, p7, p9, p10, p12, p13, p14, p17, "", p16, labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L"), ncol = 6, align = "v")

sigSNPs_boxplots <- plot_grid(p1, p2, p6, p8, p11, p15, p3, p4, p5, p7, p9, p10, p12, p13, p14, p17, p16, labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q"), ncol = 6, align = "v")
sigSNPs_boxplots
#Save plot
ggsave(plot = sigSNPs_boxplots, file = paste("sigSNPs_boxplots.pdf", sep = ""), height = 17, width = 16, path = "/project/dthavar/dilthavar/acballe/PSPPC_Phenotypes") #note that ggsave will save the last plot you

```

```{r, echo=TRUE}
###############################################################################################################
```

# ############################################################################################
#### RUN PLINK to compute LD Deay with PopLDdecay

https://github.com/BGI-shenzhen/PopLDdecay

cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs

# install
git clone https://github.com/hewm2008/PopLDdecay.git 
cd PopLDdecay
chmod 755 configure
module load gcc/6.3.0
./configure
make
mv PopLDdecay  bin/

#execute
./PopLDdecay/bin/PopLDdecay
        
#run        
./PopLDdecay/bin/PopLDdecay -InVCF /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf -MaxDist 600 -MAF 0.1 -OutStat PSPPC.filt.recode267.RC.LDdecay
zcat PSPPC.filt.recode267.RC.LDdecay.stat.gz | head
zcat PSPPC.filt.recode267.RC.LDdecay.stat.gz > PSPPC.filt.recode267.RC.LDdecay.stat

module load r/4.4.1
module load perl/5.41.3

perl ./PopLDdecay/bin/Plot_OnePop.pl -inFile PSPPC.filt.recode267.RC.LDdecay.stat.gz -output PSPPC.filt.recode267.RC.LDdecay.stat.plot

#Analyze & plot PopLDdecay results

```{r, echo=TRUE}

#WD
# setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs")
# getwd()
# 
# LDdecay <- read.table("PSPPC.filt.recode267.RC.LDdecay.stat", head = TRUE, col.names=c("Dist", "Mean_r2", "Mean_D", "Sum_r2", "Sum_D", "NumberPairs"))
# head(LDdecay)
# tail(LDdecay)
#  
# # plot LD decay
# ggplot(LDdecay, aes(x=(Dist/1000), y=Mean_r2)) + geom_line() +
#   xlab("Distance (kb)") + ylab(expression(italic(r)^2))
# 
# LDdecayplot <- ggplot(LDdecay, aes(x=Dist, y=Mean_r2, group=1)) +
#     geom_line() + 
#     theme_bw() +
#     labs(x = "Distance (kb)", y = expression(italic(r)^2))
# LDdecayplot
# 
# # Plot Mean r² vs Distance
# ggplot(LDdecay, aes(x=Dist, y=Mean_r2)) +
#   geom_line(color = "red") +
#   geom_point(color = "red") +
#   labs(
#     title = "LD Decay Curve",
#     x = "Distance (KB)",
#     y = expression(r^2)
#   ) +
#   theme_minimal(base_size = 14)
# 
# # plot LD decay
# ggplot(LDdecay, aes(x=(Dist/1000), y=Mean_r2)) + geom_line() +
#   xlab("Distance (kb)") + ylab(expression(italic(r)^2))
# 

#Save plot
# ggsave(plot = LDdecayplot, file = paste("LDdecayplot.pdf", sep = ""), height = 4, width = 6, path = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs") 
# ggsave(plot = LDdecayplot, file = paste("LDdecayplot.png", sep = ""), height = 4, width = 6, path = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs") 

```

```{r, echo=TRUE}
###############################################################################################################
```

# ############################################################################################
#### RUN PLINK to compute LD between SNPs

There is 82 single SNPs

List is save here: /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_list.txt (need to remove "s in this file)

Blink result here: /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink

#info
https://zzz.bwh.harvard.edu/plink/ld.shtml  ---> this is for previous version, some flags do not work

plink --pedmap        #Tells PLINK to use .ped and .map, use prefix. Specify .ped + .map filename prefix.
--ld-snp ${SNP_NAME}  #Tells PLINK to center LD calculations around this SNP, SNP_NAME should match the SNP ID in your .map or .bim file.
--r2-phased         
--ld-window-kb 1000   #Only compute LD between SNPs within 1000 kilobases (1 Mb) 
--ld-window 99999     #Sets the maximum number of SNPs to consider in the LD window. 99999 is a high number, effectively allowing all nearby SNPs within 1 Mb to be considered.
--ld-window-r2 0      #Minimum r² threshold to report. 0 means report all pairs regardless of LD strength (i.e., no filtering). I tested 0.2, and filter out <0.2
--out ${SNP_NAME}     #Specify prefix for output files. it creates {SNP_NAME}_LD.vcor and {SNP_NAME}_LD.log

.vcor is a variant correlation matrix file, which stores LD (r²) values between SNPs.
   
ERRORS:
•Error: Unrecognized flag ('--noweb').
•Error: --r2 has been replaced by --r2-phased and --r2-unphased.
--r2-phased computes the textbook haplotype-frequency-based r^2, and
corresponds to PLINK 1.9 --r2's behavior when the 'd', 'dprime', or
'dprime-signed' modifier was present.  --r2-unphased computes the simpler r^2
between (unphased) dosage vectors, and corresponds to how PLINK 1.x --r2
behaved without a D'-related modifier.
•SNP list as 1:48617016 instead 1_48617016

Example:
1. Run plink
plink --pedmap /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC --ld-snp 5:78533957_LD --r2-unphased --ld-window-kb 1000 --ld-window 99999 --ld-window-r2 0 --out 5:78533957_LD
2. Read plink result and comnver into csv for R
sed -e 's/\t/,/g' -e 's/   */,/g' 5:78533957_LD.vcor > 5:78533957_LD.csv 
3. Haplotyp block estimation
plink --bfile 5:78533957_LD.vcor --blocks 

                              
______________________ Run GWAS in palmetto 
sbatch LD_for_SNPs.sbatch
squeue --me
                      ______________________ LD_for_SNPs.sbatch
                      #!/bin/bash
                                                                  
                      #SBATCH --job-name LD_for_SNPs
                      #SBATCH --nodes 1
                      #SBATCH --ntasks 1
                      #SBATCH --tasks-per-node 1
                      #SBATCH --cpus-per-task 1
                      #SBATCH --mem 8gb
                      #SBATCH --time 3:00:00
                      #SBATCH --mail-type BEGIN
                      #SBATCH --mail-type END
                      #SBATCH --mail-type FAIL
                      #SBATCH --output=LD_for_SNPs.out
                      #SBATCH --error=LD_for_SNPs.err
                                            
                      module load plink/2.00_alpha_6.9
                                            
                      cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs
                                            
                      # PLINK FORMAT IS HERE /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink
                      
                      # if 10 SNPs, second number should be 10 that is 1..10
                      TMPDIR="."
                      
                      for i in {1..89}
                      do
                       echo "Working with SNP ${i}"
                       SNP_NAME=$(head -n ${i} /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_list_mod.txt | tail -n 1)
                       
                       echo "Working with SNP ${SNP_NAME}"
                       
                       plink --pedmap /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC \
                             --ld-snp ${SNP_NAME} \
                             --r2-phased \
                             --ld-window-kb 1000 \
                             --ld-window 99999 \
                             --ld-window-r2 0  \
                             --out ${SNP_NAME}_LD
                                             
                       sed -e 's/\t/,/g' -e 's/   */,/g' ${SNP_NAME}_LD.vcor > ${SNP_NAME}_LD.vcor.csv  
                      done
                      ______________________

#get plink results and put togetger is a single file

```{r, include=TRUE}

#WD
setwd("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs/")
getwd()

print("#-------------------------------POut all plink result toguether-------------------------------#")

# Load sample file to copy structure (correct types)
sample_LD <- read.csv("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs/5:78533957_LD.vcor.csv", stringsAsFactors = FALSE)
head(sample_LD)
# Rename column if needed
sample_LD <- sample_LD %>% dplyr::rename("CHROM_A" = matches("X.CHROM_A")) %>% dplyr::rename("R2" = matches("PHASED_R2"))
# Create empty dataframe with correct structure
dataLD_ALLchrs <- sample_LD[0, ]  # ← this is key, select only column names
str(dataLD_ALLchrs)
head(dataLD_ALLchrs)

#Read SNP list
SNP_LIST <- read.table("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_list.txt")
SigSNPs <- SNP_LIST$V1
#SigSNPs <- c("1:48617016")
SigSNPs

#create empty list to sabe start and end SNPs
LD_plink_start_endSNPs <- list() #create empty list
LD_plink_start_endSNPs

for (i in SigSNPs){
  print("#############################################")
  print(paste0("Processing SigSNPs: ", i))
  # Read the LD file for the current SNP
  dataLD <- read.csv(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs/",i, "_LD.vcor.csv", sep=""), head = TRUE)
  dataLD <- dataLD %>% dplyr::rename("CHROM_A" = matches("X.CHROM_A")) %>% dplyr::rename("R2" = matches("PHASED_R2"))
  print(dataLD)
  str(dataLD)
  print(dim(dataLD))
  #dataLD <- dataLD %>% dplyr::filter(R2 >= 0.4) 
  print(dataLD)
  dataLD_ALLchrs <- bind_rows(dataLD_ALLchrs,dataLD)
  
  #save start and end SNP from eacg file
  print(paste0(unique(dataLD$CHROM_A),":",min(dataLD$POS_B),"-",max(dataLD$POS_B)))
  dbLD_chr<- paste0(unique(dataLD$CHROM_A),":",min(dataLD$POS_B),"-",max(dataLD$POS_B))
  LD_plink_start_endSNPs[[i]] <- dbLD_chr
  print("#############################################")
}
#this will have all plink results stacked in a single file
head(dataLD_ALLchrs)
tail(dataLD_ALLchrs)
as.data.frame(dataLD_ALLchrs %>% group_by(CHROM_A,POS_A,ID_A) %>% dplyr::summarize(countTotal = n()))
str(dataLD_ALLchrs)
write.csv(dataLD_ALLchrs, file = "/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/R_LDBlockShow/dataLD_ALLchrs.csv", row.names = FALSE)

as.data.frame(dataLD_ALLchrs %>% dplyr::group_by(CHROM_A, CHROM_B) %>% dplyr::summarize(countTotal = n()))
as.data.frame(dataLD_ALLchrs %>% dplyr::group_by(CHROM_A, CHROM_B) %>% dplyr::filter(R2 >= 0.4) %>% dplyr::summarize(countTotal = n()))
as.data.frame(dataLD_ALLchrs %>% dplyr::filter(R2 >= 0.4) %>% dplyr::group_by(CHROM_A, POS_A) %>% dplyr::summarize(countTotal = n()))
as.data.frame(dataLD_ALLchrs %>% dplyr::filter(CHROM_A == 5) %>% dplyr::filter(R2 >= 0.4))

#Get Plink resuls and get start-end position for each SIg SNP to run LDBlockShow
#save strat an end marker of each plink file for each sign SNPs
head(LD_plink_start_endSNPs)
str(LD_plink_start_endSNPs)
dbLD_plink_db <- as.data.frame(ldply(LD_plink_start_endSNPs, data.frame))
colnames(dbLD_plink_db)[1:2] <-  c("PeakSNP", "Start_EndSNPs")
head(dbLD_plink_db)
dbLD_plink_db
str(dbLD_plink_db)
as.data.frame(dbLD_plink_db$Start_EndSNPs)
as.data.frame(unique(dbLD_plink_db$Start_EndSNPs))


# print("#-------------------------------Get HD blocks manually-------------------------------#") #----> check this because I haven check really well the results.
# # Initialize an empty data frame to store results
# LDblocks_ALLchrs <- data.frame(Chr = integer(), Pos = integer(), Head_SNP = character(), LD_Block_size = numeric(),
#                               First_LD_SNP = integer(), Last_LD_SNP = integer())
# LDblocks_ALLchrs
# 
# #Read SNP list
# SNP_LIST <- read.table("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_list.txt")
# SigSNPs <- SNP_LIST$V1
# #SigSNPs <- c("5:65228308", "1:48617016")
# SigSNPs
# 
# for (i in SigSNPs){
#   print("#############################################")
#   print(paste0("Processing SigSNPs: ", i))
#   # Read the LD file for the current SNP
#   dataLD <- read.csv(paste0("/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/LD_for_SNPs/",i, "_LD.vcor.csv", sep=""), head = TRUE)
#   dataLD <- dataLD %>% rename("CHROM_A" = matches("X.CHROM_A")) %>% rename("R2" = matches("PHASED_R2"))
#   print(dataLD)
#   # Filter for highly linked SNPs with r² > 0.8
#   ld_block <- dataLD %>% filter(R2 >= 0.4) # Highly linked SNPs, r² > 0.8 are regions are likely LD blocks.
#   print(ld_block)
# 
#   if (nrow(ld_block) == 0) {
#   cat("  No LD SNPs found for", i, "\n")
#     next
#   }
# 
#   # Assume your data frame is called df
#     Chromosome <- unique(ld_block$CHROM_A)
#     Chromosome
#     Position <- unique(ld_block$POS_A)
#     Position
#     head_SNP <- unique(ld_block$ID_A)
#     head_SNP
#     first_val <- min(ld_block$POS_B)-100000
#     first_val
#     last_val  <- max(ld_block$POS_B)+100000
#     last_val
#     size <- (last_val - first_val)-1
#     size
#     # Create a new data frame with these two values
#     summary_by_SNP <- data.frame(
#       Chr = Chromosome,
#       Pos = Position,
#       Head_SNP = head_SNP,
#       LD_Block_size = size,
#       First_LD_SNP = first_val,
#       Last_LD_SNP  = last_val)
#     # Show the result
#     print(summary_by_SNP)
#     str(summary_by_SNP)
# 
#     #out everything together
#     LDblocks_ALLchrs <- bind_rows(LDblocks_ALLchrs, summary_by_SNP)
# 
# 
#   print("#############################################")
# }
# head(LDblocks_ALLchrs)
# tail(LDblocks_ALLchrs)
# str(LDblocks_ALLchrs)

```


```{r, echo=TRUE}
###############################################################################################################

```

Once I identified LD block (manualy in excel), get the genes located inside the LD bloks

cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/R_LDBlockShow/

• We need gff:
/project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3
head /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3

grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | cut -f1 | sort -u | head
chr1LG6
chr2LG1
chr3LG5
chr4LG4
chr5LG3
chr6LG2
chr7LG7

#this code would read annotation file and will select genes from specific chrs and positions (will get the fist instance)
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 416235389 && $4 <= 416349103 {OFS="\t"; print}' | cat
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 416235389 && $4 <= 416349103 {OFS="\t"; print}' | cat

grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 25360403 && $4 <= 25557309 {OFS="\t"; print}' | cat
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 25360403 && $4 <= 25557309 {OFS="\t"; print}' | wc-l
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 25360403 && $4 <= 25557309 {OFS="\t"; print}' | cat
 
# do it - GET GENES - 67 LD BLOCKS
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "gene" && $5 >= 48617021 && $4 <= 48717017 {OFS="\t"; print}' >  AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "gene" && $5 >= 168230190 && $4 <= 168430192 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "gene" && $5 >= 221867441 && $4 <= 221967537 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "gene" && $5 >= 308298799 && $4 <= 308486547 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 1408166 && $4 <= 1409028 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 5638059 && $4 <= 5715352 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 104284193 && $4 <= 104284206 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 292234124 && $4 <= 292338589 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 292234124 && $4 <= 292338596 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 292234124 && $4 <= 292338643 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "gene" && $5 >= 320250795 && $4 <= 320250798 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 25360403 && $4 <= 25557309 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 211166775 && $4 <= 211168429 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 219740018 && $4 <= 219940020 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 239501304 && $4 <= 239590105 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 239501304 && $4 <= 239590105 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "gene" && $5 >= 434614462 && $4 <= 434714475 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 58337483 && $4 <= 58365651 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 75938721 && $4 <= 76101244 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 299349388 && $4 <= 299451819 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 416235389 && $4 <= 416335591 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 416235389 && $4 <= 416349103 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "gene" && $5 >= 432393538 && $4 <= 432460013 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 61678013 && $4 <= 61777900 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 61633165 && $4 <= 61778586 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65087309 && $4 <= 65232827 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65128307 && $4 <= 65232827 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65130720 && $4 <= 65232827 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 65132809 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68194793 && $4 <= 68194859 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68094792 && $4 <= 68194859 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68094815 && $4 <= 68194859 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68309703 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68318522 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68318539 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68323179 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 68326549 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 69819881 && $4 <= 69819934 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 74920966 && $4 <= 75120968 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 74920976 && $4 <= 75120978 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 78433946 && $4 <= 78535274 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 78433956 && $4 <= 78535274 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91271107 && $4 <= 91370972 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91254261 && $4 <= 91370972 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91271107 && $4 <= 91371027 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91254261 && $4 <= 91371027 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91271107 && $4 <= 91371044 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91254261 && $4 <= 91371044 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91271123 && $4 <= 91371116 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 91254261 && $4 <= 91371116 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 92237296 && $4 <= 92237439 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 132552034 && $4 <= 132552119 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 132552034 && $4 <= 132552119 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 138525516 && $4 <= 138544000 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 185385593 && $4 <= 185483735 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 185383675 && $4 <= 185385570 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 235813544 && $4 <= 235913609 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "gene" && $5 >= 478041316 && $4 <= 478158827 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr6LG2" && $3 == "gene" && $5 >= 102304663 && $4 <= 102304967 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr6LG2" && $3 == "gene" && $5 >= 229499298 && $4 <= 229599097 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr6LG2" && $3 == "gene" && $5 >= 229499298 && $4 <= 229599301 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr7LG7" && $3 == "gene" && $5 >= 47265878 && $4 <= 47367502 {OFS="\t"; print}' >> AllChrs.gene.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr7LG7" && $3 == "gene" && $5 >= 138068354 && $4 <= 138268356 {OFS="\t"; print}' >> AllChrs.gene.txt

# do it - GET mRNA - 67 LD BLOCKS
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "mRNA" && $5 >= 48617021 && $4 <= 48717017 {OFS="\t"; print}' >  AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "mRNA" && $5 >= 168230190 && $4 <= 168430192 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "mRNA" && $5 >= 221867441 && $4 <= 221967537 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr1LG6" && $3 == "mRNA" && $5 >= 308298799 && $4 <= 308486547 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 1408166 && $4 <= 1409028 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 5638059 && $4 <= 5715352 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 104284193 && $4 <= 104284206 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 292234124 && $4 <= 292338589 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 292234124 && $4 <= 292338596 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 292234124 && $4 <= 292338643 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr2LG1" && $3 == "mRNA" && $5 >= 320250795 && $4 <= 320250798 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 25360403 && $4 <= 25557309 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 211166775 && $4 <= 211168429 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 219740018 && $4 <= 219940020 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 239501304 && $4 <= 239590105 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 239501304 && $4 <= 239590105 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr3LG5" && $3 == "mRNA" && $5 >= 434614462 && $4 <= 434714475 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 58337483 && $4 <= 58365651 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 75938721 && $4 <= 76101244 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 299349388 && $4 <= 299451819 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 416235389 && $4 <= 416335591 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 416235389 && $4 <= 416349103 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr4LG4" && $3 == "mRNA" && $5 >= 432393538 && $4 <= 432460013 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 61678013 && $4 <= 61777900 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 61633165 && $4 <= 61778586 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65087309 && $4 <= 65232827 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65128307 && $4 <= 65232827 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65130720 && $4 <= 65232827 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65228299 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 65132809 && $4 <= 65230749 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68194793 && $4 <= 68194859 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68094792 && $4 <= 68194859 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68094815 && $4 <= 68194859 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68309703 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68318522 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68318539 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68323179 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 68326549 && $4 <= 68453819 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 69819881 && $4 <= 69819934 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 74920966 && $4 <= 75120968 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 74920976 && $4 <= 75120978 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 78433946 && $4 <= 78535274 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 78433956 && $4 <= 78535274 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91271107 && $4 <= 91370972 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91254261 && $4 <= 91370972 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91271107 && $4 <= 91371027 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91254261 && $4 <= 91371027 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91271107 && $4 <= 91371044 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91254261 && $4 <= 91371044 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91271123 && $4 <= 91371116 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 91254261 && $4 <= 91371116 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 92237296 && $4 <= 92237439 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 132552034 && $4 <= 132552119 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 132552034 && $4 <= 132552119 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 138525516 && $4 <= 138544000 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 185385593 && $4 <= 185483735 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 185383675 && $4 <= 185385570 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 235813544 && $4 <= 235913609 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $5 >= 478041316 && $4 <= 478158827 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr6LG2" && $3 == "mRNA" && $5 >= 102304663 && $4 <= 102304967 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr6LG2" && $3 == "mRNA" && $5 >= 229499298 && $4 <= 229599097 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr6LG2" && $3 == "mRNA" && $5 >= 229499298 && $4 <= 229599301 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr7LG7" && $3 == "mRNA" && $5 >= 47265878 && $4 <= 47367502 {OFS="\t"; print}' >> AllChrs.mRNA.txt
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr7LG7" && $3 == "mRNA" && $5 >= 138068354 && $4 <= 138268356 {OFS="\t"; print}' >> AllChrs.mRNA.txt

```{r, echo=TRUE}
###############################################################################################################

```

# Identify haplotype blocks

cd /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/R_LDBlockShow/

# ############################################################################################ --->> It run well in mac but DID NOT USE IT

# RUN HAPLOVIEW IN MY MAC, /Users/CaroBallen/Documents/Projects_Pulses_2025/Pea_GWAS
Download Haploview.jar from 
https://www.broadinstitute.org/haploview/downloads

it should be run as:
java -jar Haploview.jar

# ############################################################################################

#Lets use LDBlockShow - https://github.com/BGI-shenzhen/LDBlockShow
• Install
git clone https://github.com/BGI-shenzhen/LDBlockShow.git
cd LDBlockShow/
chmod 755 configure
module load gcc/6.3.0
./configure
make
mv LDBlockShow  bin/

# execute 
./LDBlockShow/bin/LDBlockSho

ls /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/R_LDBlockShow/LDBlockShow/bin/ShowLDSVG
adit first line with "#!/project/cugbf/software/perl/5.41.3/bin/perl -w" 

# ##########
#run test
./LDBlockShow/bin/LDBlockShow -InVCF ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut out -Region chr11:24100000:24200000 -OutPng -SeleVar 1
./LDBlockShow/bin/LDBlockShow -InVCF ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut out -Region chr11:24100000:24200000 -OutPng -SeleVar 2
# [-SeleVar 1] is D',[-SeleVar 2] is RR ,[-SeleVar 3] are RR and D',[-SeleVar 4] are D' and RR # the default is D'

./LDBlockShow/bin/LDBlockShow -InVCF ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut out -Region chr11:24100000:24200000 -InGWAS ./LDBlockShow/example/Example2/gwas.pvalue  -OutPng
./LDBlockShow/bin/LDBlockShow -InVCF ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut out -Region chr11:24100000:24200000 -InGWAS ./LDBlockShow/example/Example2/gwas.pvalue  -OutPng  -SeleVar 4
## you can run ShowLDSVG with more parameters to optimize the plot  ##
# ../../bin/ShowLDSVG    -InPreFix       out     -OutPut out     -InGWAS gwas.pvalue -Cutline  7  -ShowNum -PointSize 3 -OutPng

./LDBlockShow/bin/LDBlockShow -InVCF  ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut  out -InGWAS ./LDBlockShow/example/Example3/gwas.pvalue  -InGFF  ./LDBlockShow/example/Example3/In.gff -Region  chr11:24100000:24200000 -OutPng -SeleVar 1 
./LDBlockShow/bin/LDBlockShow -InVCF  ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut  out -InGWAS ./LDBlockShow/example/Example3/gwas.pvalue  -InGFF  ./LDBlockShow/example/Example3/In.gff -Region  chr11:24100000:24200000 -OutPng -SeleVar 2
_____________>>>>>

./LDBlockShow/bin/LDBlockShow -InVCF  ./LDBlockShow/example/Example1/Test.vcf.gz -OutPut  out -InGWAS ./LDBlockShow/example/Example3/gwas.pvalue  -InGFF  ./LDBlockShow/example/Example3/In.gff -Region  chr11:24100000:24200000 -OutPng -SeleVar 2 -Cutline  7

#check chromosome sizes
bcftools view -H /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf | awk '{print $1, $2}' | sort -k1,1 -k2,2n | \
awk '{ if ($1 != chr) { if (chr != "") print chr, start, prev; chr=$1; start=$2; } prev=$2 } END { print chr, start, prev }'

chr start end
1 13969 372128575
2 15328 427408574
3 158171 437508942
4 32475 446254349
5 382410 579121809
6 17541 479921918
7 271973 491299738

```{r, echo=TRUE}
## ############################################################################################
```

# file thta I need 
• We need vcf:
/project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf
bcftools query -f '%CHROM %POS\n' /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf | head
bcftools query -f '%CHROM %POS\n' /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf | tail

• We need GWAS: chr position Pvalue
/project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_chr_site_Pvalue.txt
head /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_chr_site_Pvalue.txt

• We need gff:
/project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3
head /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3

grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | cut -f1 | sort -u | head
chr1LG6
chr2LG1
chr3LG5
chr4LG4
chr5LG3
chr6LG2
chr7LG7

#this code would read annotation file and will select genes from specific chrs and positions, and rename chr to match Sig SNPs names, also from vcf file

lets get genes from region in Chr5 associates with protein and SAA 68094792-68453819

grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $4 >= 68094792 && $5 <= 68453819' | head
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $4 >= 68094792 && $5 <= 68453819 {OFS="\t"; print}' | awk -F'\t' '!seen[$4]++' | head
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $4 >= 68094792 && $5 <= 68453819 {OFS="\t"; print}' | awk -F'\t' '!seen[$4]++' | wc -l
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $3 == "mRNA" && $4 >= 68094792 && $5 <= 68453819 {OFS="\t"; print}' | awk -F'\t' '!seen[$4]++' 
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $4 >= 68094792 && $5 <= 68453819 {OFS="\t"; print}' | awk -F'\t' '!seen[$4]++' 

# do it

grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $5 >= 68094792 && $4 <= 68194859 {$1="5"; OFS="\t"; print}' > chr5LG3_genes_a.gff3
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $5 >= 68309703 && $4 <= 68453819 {$1="5"; OFS="\t"; print}' >> chr5LG3_genes_a.gff3
grep -v "^#" /project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3 | awk '$1 == "chr5LG3" && $5 >= 69819881 && $4 <= 69819934 {$1="5"; OFS="\t"; print}' >> chr5LG3_genes_a.gff3

#this colde will read significan SNPs and P-values, select by chromosome and separta by tabs

awk '$1 == "5" && $2 >= "68094792" && $2 <= "68453819" { print $1, $2, $3 }' OFS="\t" /project/dthavar/dilthavar/acballe/PSPPC_GWAS/GWAS_bayesREE/allsigsnps/unique_sigsnps_chr_site_Pvalue.txt > chr5.gwas.pvalue

#Run LDBlockShow ->  is running fine, TEST if program is working
./LDBlockShow/bin/LDBlockShow -InVCF /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf -OutPut TestCaro -Region 5:68094792-68453819 -OutPng -SeleVar 2

./LDBlockShow/bin/LDBlockShow -InVCF /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/PSPPC.filt.recode267.RC.vcf -OutPut 5:68094792-68453819 -InGWAS chr5.gwas.pvalue -InGFF chr5LG3_genes_a.gff3 -Region 5:68094792-68453819 -OutPng -OutPdf -SeleVar 2 -Cutline 6.034


```{r, include=TRUE}
sessionInfo <- sessionInfo()
sessionInfo

#clear environment
rm(list = ls())

```









































# ##############################################################################################################
# BELOW DID NOT WORK

```{r, echo=TRUE}
###############################################################################################################
```

# Identify Genes in LD blocks

## features_from_snps

https://github.com/jlboat/features_from_snps?tab=readme-ov-file

# ####################### FILES

Gene annotations (http://useast.ensembl.org/info/website/upload/gff3.html#fields):

/project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations

Pisum_sativum_v1a_genes.gff3 

Pisum_sativum_v1a_genes.gff3.sqlite

INPUT files 

/project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3

OR

/project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3.sqlite

OUTPUT PATH

cd /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/

LD Blocks with SigSNPs

PSP_SigSNPs_LD_blocks.csv

markerName,chromosome,position,start,end
S1LG6_17400543,S1LG6,17400543,17393912,17423471
S1LG6_286829090,S1LG6,286829090,286809718,286929090
S2LG1_107783965,S2LG1,107783965,107683965,107837349
S2LG1_292238595,S2LG1,292238595,292234124,292338595
S2LG1_7763861,S2LG1,7763861,7763799,7763871
S3LG5_62501066,S3LG5,62501066,62501019,62501167
S3LG5_84775906,S3LG5,84775906,84724796,84775943
S4LG4_432401765,S4LG4,432401765,432393538,432425643
S4LG4_442306235,S4LG4,442306235,442306235,442306238
S4LG4_445499049,S4LG4,445499049,445482361,445499831
S5LG3_65228308,S5LG3,65228308,65228299,65230749
S5LG3_68418453,S5LG3,68418453,68318453,68425399
S5LG3_68418523,S5LG3,68418523,68318523,68425399
S5LG3_68418618,S5LG3,68418618,68318618,68425399
S5LG3_78533947,S5LG3,78533947,78533947,78535274
S5LG3_78533957,S5LG3,78533957,78533947,78535274
S6LG2_109240779,S6LG2,109240779,109240618,109340779
S6LG2_2525216,S6LG2,2525216,2466497,2534216
S6LG2_27527145,S6LG2,27527145,27497483,27527345
S6LG2_426198885,S6LG2,426198885,426171652,426199059
S7LG7_248058260,S7LG7,248058260,247958260,248158260
S7LG7_324692136,S7LG7,324692136,324691835,324699997
S4LG4_168524733,S4LG4,168524733,168424733,168524938

# ####################### CODE 
   
## run features_from_snps.py -----> NOT WORKING

module load python/3.12.3

python features_from_snps.py -h 
 
example:
python features_from_snps.py --input GAPIT.MLM.Fe.GWAS.Results.csv --input_type gapit --output Iron_genes --gff Pisum_sativum_v1a_genes.db --distance 90 --fdr 0.05 --feature_type gene
python features_from_snps.py --input GAPIT.MLM.Fe.GWAS.Results.csv --input_type gapit --output Iron_genes --gff Pisum_sativum_v1a_genes.db --distance 90 --fdr 0.05 --feature_type gene

python features_from_snps.py --input PSP_SigSNPs_LD_blocks.csv --input_type range --output PSP_SigSNPs_LD_blocks_genes --gff Pisum_sativum_v1a_genes.gff3 --feature_type gene

python features_from_snps.py --input S1LG6_17400543.plink.ld --input_type gapit --output PSP_SigSNPs_LD_blocks_genes --gff Pisum_sativum_v1a_genes.gff3 --distance 90 --fdr 0.05 --feature_type gene


python create_db.py &

python features_from_snps.py --input PSP_SigSNPs_LD_blocks.csv --input_type range --output PSP_SigSNPs_LD_blocks_genes --gff Pisum_sativum_v1a_genes.db --feature_type gene
python features_from_snps.py --input S1LG6_17400543.plink.ld --input_type gapit --output PSP_SigSNPs_LD_blocks_genes --gff Pisum_sativum_v1a_genes.db --distance 90 --fdr 0.05 --feature_type gene


python features_from_snps.py --input PSP_SigSNPs_LD_blocks.csv --input_type range --output PSP_SigSNPs_LD_blocks_genes --gff Pisum_sativum_v1a_genes.gff3.sqlite --feature_type gene


S1LG6_17400543.plink.ld


    
python features_from_snps.py --gff Pisum_sativum_v1a_genes.gff3 --feature-type gene --input PSP_SigSNPs_LD_blocks.csv --input-type range --output PSP_SigSNPs_LD_blocks_genes


          # PLINK RESULTS will; be processed with Pathway Association Study Tool (PAST 2.0.0-rc) (https://github.com/IGBB/PAST)
          
          /project/dthavar/dilthavar/njohns9/PSPProteinGWAS/LD
          
          In palmetto2
          
          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
          
                To get started you may need to restart your current shell.
                This would reload your PATH environment variable to include
                Cargo's bin directory ($HOME/.cargo/bin).
                
                To configure your current shell, you need to source
                the corresponding env file under $HOME/.cargo.
                
                This is usually done by running one of the following (note the leading DOT):
                . "$HOME/.cargo/env"            # For sh/bash/zsh/ash/dash/pdksh
                source "$HOME/.cargo/env.fish"  # For fish
                source "$HOME/.cargo/env.nu"    # For nushell
          
          /home/acballe/.cargo/bin/cargo install --git https://github.com/IGBB/past
          
          /home/acballe/.cargo/bin/cargo install --path ./PAST/

# ############################################################################################
# run PLINK
# Haplotyp3 block estimation

module load plink/1.07

The command
plink --bfile mydata --blocks 
plink --noweb --bfile /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC2 --blocks --out PSPPC_blocks

plink --noweb --bfile /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC2 --blocks 267 --make-founders

plink --noweb --bfile /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC2 --blocks --ld-window-kb 1000


Install plink 1.9
mkdir -p plink1.9
cd plink1.9
wget https://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20201019.zip
unzip plink_linux_x86_64_20201019.zip
chmod +x plink
export PATH=$PWD:$PATH


./plink1.9/plink --bfile /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC --freq --out PSPPC_freq

./plink1.9/plink --noweb --bfile /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC --blocks no-pheno-req --make-founders --out PSPPC 
./plink1.9/plink --bfile /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC --blocks no-pheno-req --make-founders --out PSPPC 

# Prepare output file
echo -e "Block#\tCHR\tBP1\tBP2\tSNPs" > PSPPC_LD_Blocks_With_SNPs.tsv

# Loop over each block
awk 'NR>1 {print NR-1, $1, $2, $3}' PSPPC.blocks.det | while read idx chr start end; do snps=$(awk -v c=$chr -v s=$start -v e=$end '$1 == c && $4 >= s && $4 <= e {print $2}' /project/dthavar/dilthavar/acballe/PSPPC_PopStructure_Genetics/plink/PSPPC.filt.recode267.RC.bim | paste -sd "," -) echo -e "$idx\t$chr\t$start\t$end\t$snps" >> PSPPC_LD_Blocks_With_SNPs.tsv done


``` {r, echo=FALSE}

# library(rtracklayer)
# 
# 
# 
# # Read the GFF3 file into an R object
# 
# gff <- import.gff("/project/dthavar/dilthavar/ReferenceGenomes/Pisum_sativum/Cameor_v1a/annotations/Pisum_sativum_v1a_genes.gff3")
# 
# 
# 
# # Define the desired chromosome and coordinates
# 
# chromosome <- "chr1"
# 
# start <- 100000
# 
# end <- 150000
# 
# 
# 
# # Filter genes based on location
# 
# genes_in_region <- gff[gff$seqnames == chromosome & gff$start >= start & gff$end <= end]
# 
# 
# 
# # Access gene information (e.g., gene ID, name)
# 
# gene_ids <- genes_in_region$gene_id  # Assuming "gene_id" is the column containing gene identifiers


```
